### ISSUE TITLE
[2.0.4] Model::findFirst() w/ for_update leads to a PDOException

### ISSUE THREAD

**Issue by cspeak** [phalcon/cphalcon#10623]
The following code worked correctly through Phalcon 2.0.3

```php
/**
 * DailySearchStats Model
 */
class DailySearchStats extends \Phalcon\Mvc\Model 
{

   ...

    /**
     * @return DailySearchStats
     */
    public static function findFirst($parameters = [])
    {
        return parent::findFirst($parameters);
    }
 
    ...

    public static function increment()
    {
        $date = date('Y-m-d');

        $stat = DailySearchStats::findFirst([
            'conditions' => 'curdate = :theDate:',
            'bind'=>[
                'theDate' => $date
            ],
            'for_update' => true
        ]);

        if (!$stat) {
            $stat = new DailySearchStats();
            $stat->setCurdate($date);
            $stat->count = 1;
        } else {
            $stat->count++;
        }
        $stat->save();
    }
}
```

Then in the controller:

```php
...
public function searchAction(){
    DailySearchStats::increment();
}
```

However, as of 2.0.4, I get the following error

`PDOException: SQLSTATE[HY093]: Invalid parameter number: parameter was not defined`

The line with the error is actually ` return parent::findFirst($parameters);`.

If I remove `"for_update" => true`, the error disappears and everything works just fine.

I know there are plenty of other ways to perform the operation above but I'm just reporting the new bug from introduced upgrading from 2.0.3 to 2.0.4.

**Comment by andresgutierrez** [comment]
Duplicate of #10621 and it's already fixed in the 2.0.x branch

**Comment by cspeak** [comment]
Thanks for the update. I did a search before posting but looks like I barely missed it.
